GenderPayGap.WebJob/Functions/Functions.SendReminderEmails.cs

```diff
@@ -0,0 +1,191 @@
+                {"Environment", Config.IsProduction() ? "" : $"[{Config.EnvironmentName}] "}
+            };
+
+            var notifyEmail = new NotifyEmail {
+                EmailAddress = emailAddress, TemplateId = "db15432c-9eda-4df4-ac67-290c7232c546", Personalisation = personalisation
+            };
+            
+            try
+            {
+                govNotifyApi.SendEmail(notifyEmail);
+            }
+            catch (Exception ex)
+            {
+                CustomLogger.Error(
+                    "Failed to send reminder email to Notify.",
+                    new {NotifyEmail = notifyEmail, Exception = ex.Message});
+            }
+        }
+
+        private string GetOrganisationNameString(List<Organisation> organisations)
+        {
+            if (organisations.Count == 1)
+            {
+                return organisations[0].OrganisationName;
+            }
+
+            if (organisations.Count == 2)
+            {
+                return  $"{organisations[0].OrganisationName} and {organisations[1].OrganisationName}";
+            }
+
+            return $"{organisations[0].OrganisationName} and {organisations.Count - 1} other organisations";
+        }
+
+        private bool IsAfterEarliestReminder(SectorTypes sectorType)
+        {
+            return DateTime.Now > GetEarliestReminderDate(sectorType);
+        }
+
+        private bool ReminderEmailWasNotSentAfterLatestReminderDate(User user, SectorTypes sectorType)
+        {
+            ReminderEmail latestReminderEmail = _DataRepository.GetAll<ReminderEmail>()
+                .Where(re => re.UserId == user.UserId)
+                .Where(re => re.SectorType == sectorType)
+                .OrderByDescending(re => re.DateSent)
+                .FirstOrDefault();
+            
+            if (latestReminderEmail == null)
+            {
+                return true;
+            }
+
+            DateTime latestReminderEmailDate = GetLatestReminderEmailDate(sectorType);
+            return latestReminderEmail.DateSent <= latestReminderEmailDate;
+        }
+        
+        private static DateTime GetEarliestReminderDate(SectorTypes sectorType)
+        {
+            List<int> reminderEmailDays = GetReminderEmailDays();
+            int earliestReminderDay = reminderEmailDays[reminderEmailDays.Count - 1];
+
+            var deadlineDate = GetDeadlineDate(sectorType);
+            return deadlineDate.AddDays(-earliestReminderDay); 
+        }
+
+        private static DateTime GetLatestReminderEmailDate(SectorTypes sectorType)
+        {
+            List<int> reminderEmailDays = GetReminderEmailDays();
+            var deadlineDate = GetDeadlineDate(sectorType);
+            
+            IEnumerable<DateTime> reminderDates = reminderEmailDays.Select(r => deadlineDate.AddDays(-r));
+            var latestDate = reminderDates.FirstOrDefault(r => r < DateTime.Now);
+            return latestDate;
+        }
```

Comment:
It might be nice to create a method `GetReminderDates(SectorType)`, which does...
```
private static List<DateTime> GetReminderDates(SectorTypes sectorType)
{
    List<int> reminderDays = GetReminderEmailDays();
    var deadlineDate = GetDeadlineDate(sectorType);

    return reminderDays.Select(reminderDay => deadlineDate.AddDays(-reminderDay)).ToList();
}
```
...(mostly copied from `GetLatestReminderEmailDate`)

Then we can just do
`GetReminderDates(sectorType).Min()`
or
```
GetReminderDates(sectorType)
    .Where(reminderDate => reminderDate < DateTime.Now)
    .OrderBy(reminderDate => reminderDate)
    .FirstOrDefault()
```

Reply:
That sounds good, thanks